src/overmap.cpp

```diff
@@ -601,6 +599,189 @@ void reset_region_settings()
     region_settings_map.clear();
 }
 
+/*
+    Entry point for parsing "region_overlay" json objects.
+        Will loop through and apply the overlay to each of the overlay's regions.
+*/
+void load_region_overlay(JsonObject &jo)
+{
+    if (jo.has_array("regions")) {
+        JsonArray regions = jo.get_array("regions");
+
+        while (regions.has_more()) {
+            std::string regionid = regions.next_string();
+
+            if(regionid == "all") {
+                if(regions.size() != 1) {
+                    jo.throw_error("regions: More than one region is not allowed when \"all\" is used");
+                }
+
+                for(t_regional_settings_map_itr itr = region_settings_map.begin();
+                    itr != region_settings_map.end(); itr++) {
+                    apply_region_overlay(jo, itr->second);
+                }
+            }
+            else {
+                t_regional_settings_map_itr itr = region_settings_map.find(regionid);
+                if(itr == region_settings_map.end()) {
+                    jo.throw_error("region: " + regionid + " not found in region_settings_map");
+                }
+                else {
+                    apply_region_overlay(jo, itr->second);
+                }
+            }
+        }
+    }
+    else {
+        jo.throw_error("\"regions\" is required and must be an array");
+    }
+}
+
+void apply_region_overlay(JsonObject &jo, regional_settings &region)
+{
+    if (jo.has_member("default_oter")) {
+        jo.read("default_oter", region.default_oter);
+    }
+
+    if (jo.has_object("default_groundcover")) {
+        JsonObject jio = jo.get_object("default_groundcover");
+
+        if (jio.has_member("primary")) {
+            jio.read("primary", region.default_groundcover_str->primary_str);
+        }
+        if (jio.has_member("secondary")) {
+            jio.read("secondary", region.default_groundcover_str->secondary_str);
+        }
+        if (jio.has_member("ratio")) {
+            jio.read("ratio", region.default_groundcover.chance);
+        }
+    }
+
+    if (jo.has_member("num_forests")) {
+        jo.read("num_forests", region.num_forests);
+    }
+    if (jo.has_member("forest_size_min")) {
```

Comment:
Those checks here are completely pointless. `JsonObject::read` does them anyway and if the member does not exists, nothing is loaded. Just calling `jo.read( ... )` is sufficient and will reduce the code to 1/3.